### PR TITLE
Add unit tests for `coriolisclient.cli`: `shell`, `utils` and `formatter` modules

### PR DESCRIPTION
**Pull request by Cristi1324**
Add unit tests for the following modules:
- `coriolisclient.cli.formatter`
- `coriolisclient.cli.shell`
- `coriolisclient.cli.utils`

### CODE REVIEW

File: coriolisclient/cli/utils.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def get_option_value_from_args(args, option_name, error_on_no_value=True):
         with file_arg as fin:
             raw_value = fin.read()
 
-    if not value and raw_value:
+    if raw_value:
```
Comment (Dany9966) [inline coriolisclient/cli/utils.py:178]: Good catch!

File: coriolisclient/tests/cli/test_formatter.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+
+@ddt.ddt
+class EntityFormatterTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Entity Formatter."""
+
+    def setUp(self):
+        def _get_formatted_data(self, obj):
+            return mock.sentinel.data
+        setattr(formatter.EntityFormatter, '_get_formatted_data',
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Instead of doing this, let's use the `EntityFormatter` as it was supposed to be used, and that is by creating a mock class that inherits from this tested mixin class. You just need to "implement" the `_get_formatted_data` method and then you can test the rest of the methods as below.

File: coriolisclient/tests/cli/test_formatter.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    @mock.patch.object(formatter.EntityFormatter, '_get_generic_data')
+    def test_list_objects(
+        self,
+        mock_get_generic_data,
+        mock_get_sorted_list,
+        mock_get_generic_columns,
+    ):
+        obj_list = ["obj2", "obj1"]
+        mock_get_sorted_list.return_value = ["obj1", "obj2"]
+        mock_get_generic_data.side_effect = [
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: If you implement `_get_formatted_data` for your test class, you won't need to mock this, and also you wouldn't have to use `next()` in results (especially if you just return the passed obj). Let's give it a simpler look please

File: coriolisclient/tests/cli/test_formatter.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        obj = mock.Mock()
+
+        result = self.format._get_generic_data(obj)
+
+        self.assertEqual(
+            mock.sentinel.data,
+            result
+        )
+
+    def test_get_generic_columns(self):
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py:67]: Also set a test list to `columns` attribute to your test class, so you can test this method properly

File: coriolisclient/tests/cli/test_formatter.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        result = self.format._get_generic_columns()
+
+        self.assertEqual(
+            mock.sentinel.columns,
+            result
+        )
+
+    def testget_formatted_entity(self):
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Please fix this typo

File: coriolisclient/tests/cli/test_shell.py [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+                    create_keystone_session_args,
+                    auth_type=data["auth_type"]
+                )
+                mock_Client.assert_called_once_with(
+                    session=mock_create_keystone_session.return_value,
+                    **call_args
+                )
+        else:
+            self.assertRaises(  # noqa: H202
+                Exception,
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please change this Exception into a `CoriolisException` (you'll find it in the `exceptions` module), both here and in the original code. Then, you could also remove the H202 comment.

File: coriolisclient/tests/cli/test_shell.py [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+            )
+            mock_build_kwargs_based_on_version.assert_called_once_with(
+                args,
+                data.get("api_version", None)
+            )
+            mock_auth.assert_called_once_with(**expected_kwargs)
+            if data.get("err_msg", False):
+                mock_stderr.write.assert_called_once()
+            else:
+                mock_stderr.write.assert_not_called()
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: This is not necessary, please remove

File: coriolisclient/tests/cli/test_shell.py [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+                raise_exc=data.get("raise_exc", False)
+            )
+
+            self.assertEqual(
+                expected_result,
+                result
+            )
+        else:
+            self.assertRaises(  # noqa: H202
+                Exception,
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please also change these to `CoriolisException`

File: coriolisclient/tests/cli/test_shell.py [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+        mock_Client,
+        mock_create_keystone_session
+    ):
+        args = CustomMock()
+        args.os_identity_api_version = mock.sentinel.os_identity_api_version
+        for key, value in data.get("args", {}).items():
+            setattr(args, key, value)
+        endpoint_filter_kwargs = {"filter_arg": "mock_filter_arg"}
+        mock_get_endpoint_filter_kwargs.return_value = endpoint_filter_kwargs
+        call_args = data.get("call_args", None)
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Pro tip: unless set, `get()` method's `value` argument always defaults to `None`, you don't need to explicitly set it

File: coriolisclient/tests/cli/test_shell.py [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+            call_args.update(endpoint_filter_kwargs)
+
+            result = self.coriolis.create_client(args)
+
+            self.assertEqual(
+                mock_Client.return_value,
+                result
+            )
+
+            if create_keystone_session_args is None:
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: I'd prefer if you'd change this condition to `if args.no_auth:`
Makes more sense, because if auth is otherwise set, a session needs to be created, hence `create_keystone_session` is called

File: coriolisclient/tests/cli/test_shell.py [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+        self.assertEqual(
+            mock_build_option_parser.return_value,
+            result
+        )
+        mock_build_option_parser.assert_called_once_with(
+            mock.sentinel.description, mock.sentinel.version, None)
+        mock_register_session_argparse_arguments.assert_called_once_with(
+            mock_build_option_parser.return_value)
+
+    @mock.patch.object(os, 'environ')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please actually use `os.environ` instead of mocking it: https://docs.python.org/3/library/os.html#os.environ

File: coriolisclient/tests/cli/test_utils.py [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+        },
+        {
+            "global_scripts": None,
+            "instance_scripts": ["linux='invalid/file/path'"],
+            "expected_result": None
+        },
+    )
+    def test_compose_user_scripts(self, data):
+        global_scripts = data["global_scripts"]
+        global_scripts = data["global_scripts"]
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: Please remove duplicate line

File: coriolisclient/tests/cli/test_utils.py [hunk 13]
```diff
@@ -0,0 +1,311 @@
...
+            "instance_scripts": None,
+            "expected_result":
+            {
+                "global": {},
+                "instances": {}
+            }
+        },
+        {
+            "global_scripts": ["linux="],
+            "instance_scripts": ["linux="],
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: Please don't input os_type as instance. It should be something like `instance_1`

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolisclient/tests/cli/test_formatter.py [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+        obj_list = ["obj2", "obj1"]
+
+        result = self.format._get_sorted_list(obj_list)
+
+        self.assertEqual(
+            obj_list,
+            result
+        )
+
+    @mock.patch.object(formatter.EntityFormatter, '_get_generic_columns')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: You don't need to mock this. This just returns `self.columns`, which you have at `TestEntityFormattter.columns`.

File: coriolisclient/tests/cli/test_formatter.py [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+    def test_list_objects(
+        self,
+        mock_get_generic_columns,
+    ):
+        obj_list = ["obj2", "obj1"]
+        mock_get_generic_columns.return_value = mock.sentinel.columns
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Please remove this.

File: coriolisclient/tests/cli/test_formatter.py [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+
+        result = self.format.list_objects(obj_list)
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Save the result as `columns, data`, so that it's easier to check for equality. Basically `columns` should be `self.columns`, and `data` should be `obj_list`

File: coriolisclient/tests/cli/test_formatter.py [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+        self.assertEqual(
+            obj,
+            result
+        )
+
+    def test_get_generic_columns(self):
+        result = self.format._get_generic_columns()
+
+        self.assertEqual(
+            ["column_1", "column_2"],
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: You can replace this with `self.columns`

File: coriolisclient/tests/cli/test_formatter.py [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+            result
+        )
+
+    @mock.patch.object(formatter.EntityFormatter, '_get_generic_columns')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Please stop mocking this. It just returns `self.columns`. It's also not even called in `get_formatted_entity`

File: coriolisclient/tests/cli/test_formatter.py [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+    def test_get_formatted_entity(self, mock_get_generic_columns):
+        obj = mock.Mock()
+        mock_get_generic_columns.return_value = mock.sentinel.columns
+
+        result = self.format.get_formatted_entity(obj)
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: You can also save the result in `columns, data` tuple, so it's easier to read the equality asserts.

File: coriolisclient/tests/cli/test_formatter.py [hunk 20]
```diff
@@ -0,0 +1,145 @@
...
+
+        self.assertEqual(
+            (["column_1", "column_2"], obj),
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: You can use `self.columns`.

File: coriolisclient/tests/cli/test_formatter.py [hunk 21]
```diff
@@ -0,0 +1,145 @@
...
+            "max_value": None,
+            "percent_format": "{:.0f}%",
+            "expected_result": None
+        }
+    )
+    def test_get_percent_string(self, data):
+        result = self.format._get_percent_string(
+            data["current_value"],
+            data["max_value"],
+            data["percent_format"]
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py]: Please add keyword to arg: 
```suggestion
            percent_format=data["percent_format"]
```

File: coriolisclient/tests/cli/test_formatter.py [hunk 22]
```diff
@@ -0,0 +1,145 @@
...
+        )
+
+        self.assertEqual(
+            data["expected_result"],
+            result
+        )
+
+    @mock.patch.object(formatter.EntityFormatter, '_get_percent_string')
+    def test_format_progress_update(self, mock_get_percent_string):
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_formatter.py:124]: Please add a case without `percent_string`

File: coriolisclient/tests/cli/test_shell.py [hunk 23]
```diff
@@ -0,0 +1,476 @@
...
+            self.assertEqual(
+                expected_result,
+                result
+            )
+        else:
+            self.assertRaises(
+                exceptions.CoriolisException,
+                self.coriolis.check_auth_arguments,
+                args,
+                data.get("api_version", None),
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please add the keyword to argument:
```suggestion
                api_version=data.get("api_version", None),
```

File: coriolisclient/tests/cli/test_shell.py [hunk 24]
```diff
@@ -0,0 +1,476 @@
...
+class CoriolisTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Shell App."""
+
+    def setUp(self):
+        super(CoriolisTestCase, self).setUp()
+        self.coriolis = shell.Coriolis()
+
+    @ddt.data(
+        {
+            "args": {
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: This doesn't really test the edge-cases of the method. The less args you pass, the better (since `any()` combination is accepted). Please come up with a more comprehensive data, and maybe also do it in a separate yaml file. 
Stuff like passing 'v3' without args, passsing 'v3' only with `args.os_project_id`. Passing 'v3' only with `args.os_project_name`. Passing 'v3' with `args.os_project_name` and `args.os_project_domain_name`, etc.

File: coriolisclient/tests/cli/test_shell.py [hunk 25]
```diff
@@ -0,0 +1,476 @@
...
+                "os_project_domain_name": "new_mock_os_project_domain_name",
+                "os_project_domain_id": "new_mock_os_project_domain_id",
+            },
+            "auth_fun": 'keystoneauth1.identity.v3.Password',
+            "err_msg": True
+        }
+    )
+    @mock.patch.object(session, 'Session')
+    @mock.patch.object(shell.Coriolis, 'build_kwargs_based_on_version')
+    @mock.patch.object(shell.Coriolis, 'check_auth_arguments')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please remove these mocks. These only checks and edit dict values, don't really need mocking.

File: coriolisclient/tests/cli/test_shell.py [hunk 26]
```diff
@@ -0,0 +1,476 @@
...
+        mock_Session
+    ):
+        args = CustomMock()
+        for key, value in data.get("args", {}).items():
+            setattr(args, key, value)
+        kwargs = data.get("kwargs", {})
+        ret_args = data.get("args", {})
+        mock_build_kwargs_based_on_version.return_value = ret_args.copy()
+        expected_kwargs = ret_args.copy()
+        expected_kwargs.update(kwargs.copy())
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: If removing mocks, you will have to explicitly insert the expected kwargs. I feel like this is overall better for these tests, more accurate to the given code, so please do so.

File: coriolisclient/tests/cli/test_shell.py [hunk 27]
```diff
@@ -0,0 +1,476 @@
...
+                "os_project_domain_id": "mock_os_project_domain_id",
+            },
+            "kwargs": {
+                "os_project_domain_name": "new_mock_os_project_domain_name",
+                "os_project_domain_id": "new_mock_os_project_domain_id",
+            },
+            "auth_fun": 'keystoneauth1.identity.v3.Password',
+            "err_msg": True
+        }
+    )
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please also input data in a separate yaml file.

File: coriolisclient/tests/cli/test_shell.py (hunk 25 above)
Comment (Dany9966) [reply coriolisclient/tests/cli/test_shell.py]: It's fine and recommneded to mock the auth and session methods tho. Pleae keep those and properly assert what gets passed to those calls.

File: coriolisclient/tests/cli/test_shell.py [hunk 28]
```diff
@@ -0,0 +1,476 @@
...
+                "user_id": "mock_os_user_id",
+                "username": "mock_os_username",
+            },
+            "auth_type": "password"
+        },
+        {}
+    )
+    @mock.patch.object(shell.Coriolis, 'create_keystone_session')
+    @mock.patch.object(client, 'Client')
+    @mock.patch.object(shell.Coriolis, '_get_endpoint_filter_kwargs')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please remove this mock. It'll make things easier.

File: coriolisclient/tests/cli/test_shell.py [hunk 29]
```diff
@@ -0,0 +1,476 @@
...
+                "endpoint": "mock_endpoint"
+            }
+        },
+        {
+            "args": {
+                "no_auth": "mock_no_auth",
+                "endpoint": "mock_endpoint",
+                "os_tenant_id": "mock_os_tenant_id"
+            },
+            "call_args": {
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Can you please rename this to "expected_client_kwargs"?

File: coriolisclient/tests/cli/test_shell.py [hunk 30]
```diff
@@ -0,0 +1,476 @@
...
+            "args": {
+                "os_auth_token": "mock_os_auth_token",
+                "os_auth_url": "mock_os_auth_url",
+                "os_tenant_id": "mock_os_tenant_id",
+                "endpoint": "mock_endpoint"
+            },
+            "call_args": {
+                "endpoint": "mock_endpoint"
+            },
+            "create_keystone_session_args": {
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Can you please rename this also? Maybe something like "expected_ks_session_kwargs"

File: coriolisclient/tests/cli/test_shell.py [hunk 31]
```diff
@@ -0,0 +1,476 @@
...
+                "username": "mock_os_username",
+            },
+            "auth_type": "password"
+        },
+        {}
+    )
+    @mock.patch.object(shell.Coriolis, 'create_keystone_session')
+    @mock.patch.object(client, 'Client')
+    @mock.patch.object(shell.Coriolis, '_get_endpoint_filter_kwargs')
+    def test_create_client(
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py:223]: Also you don't have to do all the logic to setup the expected kwargs. Just set it up in the test data, call it, assert it. 

Here's a basic example of how it should go:
```python3
    @mock.patch.object(shell.Coriolis, 'create_keystone_session')
    def test_create_client_1(self, mock_create_ks_session):
        args = CustomMock()
        args.os_auth_token = "auth_token"
        args.os_auth_url = "auth_url"
        args.endpoint = "endpoint"
        args.os_tenant_id = "tenant_id"
        args.os_identity_api_version = "3"

        result = self.coriolis.create_client(args)

        self.assertEqual(result.session, mock_create_ks_session.return_value)

        expected_session_kwargs = {
            "args": args,
            "api_version": args.os_identity_api_version,
            "kwargs_dict": {"auth_url": args.os_auth_url,
                            "token": args.os_auth_token},
            "auth_type": "token"
        }
        mock_create_ks_session.assert_called_with(**expected_session_kwargs)
```
Ofc you having predefined data, it would look a bit different, but the "spirit" of the test should be the same. Only mock the stuff that really needs mocking. By adding endpoint filters you can also check that the filter was applied on the client object, etc. 
The filter example should only be ONE case, intended to double-check that the method is called and does affect the client object. All other filter related cases should occur when unit testing the filtering method.

File: coriolisclient/tests/cli/test_shell.py [hunk 32]
```diff
@@ -0,0 +1,476 @@
...
+                    **call_args
+                )
+        else:
+            self.assertRaises(
+                exceptions.CoriolisException,
+                self.coriolis.create_client,
+                args
+            )
+
+    def test_build_kwargs_based_on_version(self):
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: Please add a case for `api_version='3'`, and also add the `api_version=api_version` kwarg name to the calls.

Comment (Dany9966) [reply coriolisclient/tests/cli/test_shell.py]: It'd be more apprecieted if you'd use `ddt` here please

File: coriolisclient/tests/cli/test_shell.py [hunk 33]
```diff
@@ -0,0 +1,476 @@
...
+            mock.sentinel.description, mock.sentinel.version, None)
+        mock_register_session_argparse_arguments.assert_called_once_with(
+            mock_build_option_parser.return_value)
+
+    def test_env(self):
+        result = self.coriolis._env(
+            "HOME", mock.sentinel.default)
+
+        self.assertEqual(
+            os.environ.get("HOME"),
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: You're basically mimicking the method here. Please instead set a random env value and check that the method returns it. (using `os.environ`) 
```python3
os.environ["TEST_ENV"] = "test_value"
```

File: coriolisclient/tests/cli/test_shell.py [hunk 34]
```diff
@@ -0,0 +1,476 @@
...
+            (
+                logging.getLogger("requests").level,
+                logging.getLogger("keystoneclient").level
+            )
+        )
+
+    @mock.patch.object(shell.Coriolis, 'run')
+    @mock.patch.object(commandmanager, 'CommandManager')
+    @mock.patch.object(app.App, '__init__')
+    @mock.patch.object(shell, '_setup_logging')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: You can remove all these mocks, only mock the `shell.Coriolis` class.

File: coriolisclient/tests/cli/test_utils.py [hunk 35]
```diff
@@ -0,0 +1,310 @@
...
+class UtilsTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Utils."""
+
+    def test_add_storage_mappings_arguments_to_parser(self):
+        parser = argparse.ArgumentParser()
+
+        utils.add_storage_mappings_arguments_to_parser(parser)
+        args = parser.parse_args(
+            ['--storage-backend-mapping', 'mock_source=mock_destination',
+             '--disk-storage-mapping', 'mock_disk_id=mock_destination'])
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: Please add a `default_storage_backend` also. Great test btw, I love it!

File: coriolisclient/tests/cli/test_utils.py [hunk 36]
```diff
@@ -0,0 +1,310 @@
...
+            ),
+            result
+        )
+
+        result = utils.parse_storage_mappings(None)
+
+        self.assertEqual(
+            (None, {}, {}),
+            result
+        )
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: Please move this into a separate test method, call it `test_parse_storageg_mappings_none` or something like that.

File: coriolisclient/tests/cli/test_utils.py [hunk 37]
```diff
@@ -0,0 +1,310 @@
...
+        self.assertEqual(
+            '{\n  "key1": "value1",\n  "key2": "value2"\n}',
+            result
+        )
+
+        obj = mock.Mock()
+        obj.prop_name.to_dict.return_value = \
+            {"key1": "value1", "key2": "value2"}
+
+        result = utils.format_json_for_object_property(obj, "prop_name")
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py:126]: Please move all the different calls in a different test method.

File: coriolisclient/tests/cli/test_utils.py [hunk 38]
```diff
@@ -0,0 +1,310 @@
...
+        result = utils.validate_uuid_string(
+            "123456789ABCDEF")
+
+        self.assertEqual(
+            False,
+            result
+        )
+
+    def test_add_args_for_json_option_to_parser(self):
+        parser = mock.Mock()
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: I really loved the approach of using an empty `ArgumentParser` to test these kinds of methods. Can you please also do that here?

File: coriolisclient/tests/cli/test_utils.py [hunk 39]
```diff
@@ -0,0 +1,310 @@
...
+                ValueError,
+                utils.compose_user_scripts,
+                global_scripts,
+                instance_scripts
+            )
+
+    def test_compose_user_scripts_from_file(self):
+        script_path = os.path.dirname(os.path.realpath(__file__))
+        script_path = os.path.join(script_path, 'data')
+        script_path = os.path.join(script_path, 'user_scripts.yml')
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_utils.py]: You can simplify this by:
```suggestion
        script_path = os.path.join(script_path, 'data/user_scripts.yml')
```

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolisclient/tests/cli/test_shell.py [hunk 40]
```diff
@@ -0,0 +1,444 @@
...
+        {
+            "args": {
+                "no_auth": "mock_no_auth",
+                "os_auth_url": "mock_os_auth_url"
+            }
+        },
+        {
+            "args": {
+                "no_auth": "mock_no_auth",
+                "os_auth_url": "mock_os_auth_url"
```
Comment (Dany9966) [inline coriolisclient/tests/cli/test_shell.py]: These two cases seem to be the same.

**Review by Dany9966 — APPROVED** [review]
